module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp

```diff
@@ -17,6 +19,10 @@ namespace gui
 {
     class PhoneLockedWindow : public AppWindow
     {
+      private:
+        std::shared_ptr<WallpaperClock> clockWallpaper         = nullptr;
+        std::unique_ptr<WallpaperPresenter> wallpaperPresenter = nullptr;
```

Comment:
pass a generic/non generic presenter from the application and consider using reference then